LayoutTests/http/tests/paymentrequest/paymentrequest-merchantCategoryCode.https.html

```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Tests for providing `merchantCategoryCode` as part of `ApplePayRequestBase`.</title>
+<script src="/js-test-resources/ui-helper.js"></script>
+<script src="/resources/payment-request.js"></script>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+    setup({ explicit_done: true, explicit_timeout: true });
+    test((test) => {
+        const method = validPaymentMethod();
+        method.data.features = ["paymentRequestMerchantCategoryCode"];
+        new PaymentRequest([method], validPaymentDetails());
+    }, "Should have a feature for `paymentRequestMerchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const request = new PaymentRequest([validPaymentMethod()], validPaymentDetails());
+        
+        request.addEventListener("merchantvalidation", (event) => {
+            event.complete({ });
+        });
+        
+        const shippingPromise = new Promise(resolve => {
+            request.addEventListener("shippingaddresschange", (event) => {
+                assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after an update");
+                event.updateWith({ });
+                internals.mockPaymentCoordinator.acceptPayment();
+                
+                resolve();
+            });
+        });
+        
+        const response = await request.show();
+        
+        await shippingPromise;
+        
+        assert_equals(internals.mockPaymentCoordinator.merchantCategoryCode, null, "check that the `merchantCategoryCode` is still `null` after the payment is accepted");
+        
+        await response.complete("success");
+    }, "Should not have a default value for `merchantCategoryCode`.");
+    
+    user_activation_test(async (test) => {
+        const method = validPaymentMethod();
+        method.data.merchantCategoryCode = "0739";
```

Comment:
It might be good to have a comment as to what "0739" means (if anything?). And maybe a link to where it's defined if it's meaningful.

Reply:
Also, what happens if one provides a bogus method.data.merchantCategoryCode? would that reject? Should there be a test for that?